### PR TITLE
[backend/frontend] Categorize payload by architecture

### LINKED ISSUE #1189: Categorize payload by architecture
## Use case

Today we categorize payloads by platform but we need to apply another level of granularity by binary/archi 

Example: Windows ARM vs Windows  x64

### PR DESCRIPTION
**Pull request by isselparra**
### Proposed changes

-  Give the opportunity for a user to declare the compatible architecture of an Executable Payload.

### Related issues

* Closes #1189 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/migration/V3_44__Add_column_executable_arch.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Component
+public class V3_44__Add_column_executable_arch extends BaseJavaMigration {
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    Connection connection = context.getConnection();
+    Statement statement = connection.createStatement();
+    statement.execute("ALTER TABLE payloads ADD executable_arch varchar(255);");
+    statement.execute("UPDATE payloads SET executable_arch = 'x86_64' WHERE payload_type ='Executable';");
+  }
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/migration/V3_44__Add_column_executable_arch.java:19]: Maybe you can try use check to verify if payload is executable so executable_arch is not null?

File: openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        input.setExecutableArch(Endpoint.PLATFORM_ARCH.x86_64);
+        input.setExecutableFile(EXECUTABLE_FILE.getId());
+
+        mvc.perform(post(PAYLOAD_URI)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(asJsonString(input)))
+            .andExpect(status().is2xxSuccessful())
+            .andExpect(jsonPath("$.executable_arch").value("x86_64"));
+    }
+}
```
Comment (savacano28) [inline openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java:68]: Make you could add other test without arch?

File: openbas-model/src/main/java/io/openbas/database/model/Executable.java [hunk 3]
```diff
@@ -26,6 +28,13 @@ public class Executable extends Payload {
   @JsonProperty("executable_file")
   private Document executableFile;
 
+  @Queryable(filterable = true, searchable = true)
+  @Column(name = "executable_arch")
+  @JsonProperty("executable_arch")
+  @Enumerated(EnumType.STRING)
+  @NotNull
+  private Endpoint.PLATFORM_ARCH executableArch;
+
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/model/Executable.java:37]: Maybe extract this Platform_arch enum in other class?

**Review by savacano28 — APPROVED** [review]

File: openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java (hunk 2 above)
Comment (isselparra) [reply openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java:89]: Done

File: openbas-api/src/main/java/io/openbas/migration/V3_44__Add_column_executable_arch.java (hunk 1 above)
Comment (isselparra) [reply openbas-api/src/main/java/io/openbas/migration/V3_44__Add_column_executable_arch.java:19]: Discussed together, I'm not going to make this change. 

For me its better to add this kind of validation on the api/service. I included a validation in `PayloadApi`

File: openbas-model/src/main/java/io/openbas/database/model/Executable.java (hunk 3 above)
Comment (isselparra) [reply openbas-model/src/main/java/io/openbas/database/model/Executable.java:37]: I'm merging without this refacto

**Pull request merged**